src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -20833,6 +20833,46 @@ End Module
     'Public Sub M1(Of Integer, (Integer, Integer))(a As ((Integer, Integer), Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, Integer, (Integer, Integer)))': Not most specific.").WithLocation(7, 9)
                 )
         End Sub
+
+        <Fact>
+        <WorkItem(21785, "https://github.com/dotnet/roslyn/issues/21785")>
+        Public Sub TypelessTupleInArrayInitializer()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Module Module1
+    Private mTupleArray As (X As Integer, P As System.Func(Of Byte(), Integer))() = {
+            (X:=0, P:=Nothing), (X:=0, P:=AddressOf MyFunction)
```

Comment:
>(X:=0, P:=Nothing), (X:=0, P:=AddressOf MyFunction) [](start = 12, length = 51)

It would be good to add tests for scenarios when result of the inference for the array element type is important, both success and failure scenarios.